pkg/package-server/provider/registry.go

```diff
@@ -131,43 +157,176 @@ func (p *RegistryProvider) syncCatalogSource(obj interface{}) (syncError error)
 		return
 	}
 
-	key := sourceKey{source.GetName(), source.GetNamespace()}
-	client, ok := p.getClient(key)
-	if ok && source.Status.RegistryServiceStatus.ServiceName != "" {
-		logger.Info("update detected, attempting to reset grpc connection")
-		client.conn.ResetConnectBackoff()
+	address := source.Address()
+	logger = logger.WithField("address", address)
+
+	key := resolver.CatalogKey{
+		Namespace: source.GetNamespace(),
+		Name:      source.GetName(),
+	}
+	if sourceMeta := p.sources.GetMeta(key); sourceMeta != nil && sourceMeta.Address == address {
+		// If the address hasn't changed, don't bother creating a new source
+		logger.Debug("catalog address unchanged, skipping source creation")
```

Comment:
Can’t the pod be updated with new content without the address changing?

Reply:
I'm assuming that, at minimum, the gRPC connection changes state when catalog content is updated. That's probably a bad assumption in general, but I believe it's the behavior of all current registry implementations.